geospaas_harvesting/ingesters.py

```diff
@@ -310,35 +319,65 @@ def _get_xml_namespace(self, root):
             self.LOGGER.warning('Could not find XML namespace while reading DDX metadata')
         return namespace_prefix
 
-    def _extract_global_attributes(self, root):
-        """Extracts the global attributes of a dataset from a DDX document"""
+    def _extract_attributes(self, root):
+        """
+        Extracts the global or specific attributes of a dataset or specific ones from a DDX document
+
+        "x_path_global" is pointing to the 'NC_GLOBAL' part of response of the DDX document to
+        obtain general information.
+        "x_path_specific" is used to extract the dataset parameter names from the DDX document.
+        """
         self.LOGGER.debug("Getting the dataset's global attributes.")
         namespaces = {'default': self._get_xml_namespace(root)}
-        global_attributes = {}
-        for attribute in root.findall(
-                f"./default:Attribute[@name='{self.GLOBAL_ATTRIBUTES_NAME}']/default:Attribute",
-                namespaces):
-
-            global_attributes[attribute.get('name')] = attribute.find(
+        extracted_attributes = {}
+        x_path_global = "./default:Attribute[@name='NC_GLOBAL']/default:Attribute"
+        x_path_specific = "./default:Grid/default:Attribute[@name='standard_name']"
+        # finding the global metadata
+        for attribute in root.findall(x_path_global, namespaces):
+            extracted_attributes[attribute.get('name')] = attribute.find(
                 "./default:value", namespaces).text
-
-        return global_attributes
+        # finding the parameters of the dataset that are declared in
+        # the online source (specific metadata)
+        # The specific ones are stored in 'raw_dataset_parameters' part of
+        # the returned dictionary("extracted_attributes")
+        extracted_attributes['raw_dataset_parameters'] = list()
+        for attribute in root.findall(x_path_specific, namespaces):
+            extracted_attributes['raw_dataset_parameters'].append(
+                attribute.find("./default:value", namespaces).text)
+        # removing the "latitude" and "longitude" from
+        # the 'raw_dataset_parameters' part of the dictionary
+        if 'latitude' in extracted_attributes['raw_dataset_parameters']:
+            extracted_attributes['raw_dataset_parameters'].remove('latitude')
+        if 'longitude' in extracted_attributes['raw_dataset_parameters']:
+            extracted_attributes['raw_dataset_parameters'].remove('longitude')
+        return extracted_attributes
+
+    def prepare_url(self, url):
+        """
+        Converts the downloadable link into the link for reading meta data. In all cases (usage of
+        any kind of crawler), regardless of class of used crawler, this
+        method results in a url that ends with '.ddx' which will be used in further steps
+        of ingestion.
```

Comment:
No need to say the same thing three times :)
```suggestion
        Converts the downloadable link into the link for reading meta data. In all cases,
        this method results in a url that ends with '.ddx' which will be used in further steps
        of ingestion.
```

Reply:
Done in the next commit